lib/nostrum/struct/snowflake.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Nostrum.Struct.Snowflake do
+  @moduledoc """
+  Functions that work on Snowflakes.
+
+  Snowflakes are 64-bit unsigned integers used to represent discord 
+  object ids. In JSON, they are typically represented as strings due 
+  to some languages not being able to represent such a large number.
+  
+  Nostrum represents snowflakes as integers since Elixir allows us 
+  to represent large integers.
+  """
+
+  @type external_snowflake :: String.t
```

Comment:
I think I'd prefer we move some of the documentation in the `moduledoc` to `typedoc`s.

Reply:
Latest commit has reflected these changes. They look weird to me, but it is likely because of how I refactored the documentation.